x/bank/app_test.go

```diff
@@ -191,7 +229,7 @@ func TestMsgMultiSendWithAccounts(t *testing.T) {
 	baseApp := s.App.BaseApp
 	ctx := baseApp.NewContext(false)
 
-	require.NoError(t, testutil.FundAccount(ctx, s.BankKeeper, addr1, sdk.NewCoins(sdk.NewInt64Coin("foocoin", 67))))
+	require.NoError(t, testutil.FundAccount(ctx, s.BankKeeper, addr1, sdk.NewCoins(sdk.NewInt64Coin("matic", 68*feeAmount))))
```

Comment:
This doesn't seem to make sense...why change this from 67 to 68? Additionally why is this an in subsequent tests the original amount multiplied by the fee amount instead of just incrementing?

Reply:
This is because we deduct a certain amount of fee in MATIC (`10^15 MATIC`) in the `antehandler` while processing txs (see [here](https://github.com/0xPolygon/cosmos-sdk/pull/3/files#diff-b618e5d1a7bdaf97e1e5ddc6b10a2c8a907e0a1d2be54a2eee320320e69a5908R16-R25)).  I tweaked the numbers to fit the tests.

Reply:
It is to match the numbers in `expectedBalances` in the `make a valid tx` test case. We could have also tweaked `expectedBalances`.

Reply:
LGTM, will leave to @glnro for eventual resolution

Reply:
I would make sure to put a note here as to why there's missing tokens, or at least in the test verify that those fee tokens went to the appropriate destination. Looks like 1matic is missing after the multisend, it would be clearer to check the feecollector balance has increased or something (as I assume its the fee collector this has been sent to)

Reply:
That's a good point. I'll take a look.

Reply:
Updated the tests [46dd21b](https://github.com/0xPolygon/cosmos-sdk/pull/5/commits/46dd21b382fdf34785210c99c5bed9c0568fd504). 

However, as previously reported, there are some inconsistencies in `fee_collector`'s account balance. This is primarily due to the following:

- `x/mint`'s begin blocker does some calculations on how many coins to mint (based on the supply, inflation, etc.) and [sends it to the ](https://github.com/0xPolygon/cosmos-sdk/blob/507ec1de34858aed7872e60a096f043a975ab3c9/x/mint/abci.go#L56)fee_collector .

- `x/distribution`'s begin blocker [transfers all the accumulated fees to itself](https://github.com/0xPolygon/cosmos-sdk/blob/507ec1de34858aed7872e60a096f043a975ab3c9/x/distribution/keeper/allocation.go#L21-L26) and then does some other processing related to rewards calculation etc.

I have tinkered the tests keeping these in mind. But the better alternative here it seems is to make `simapp` to resemble heimdall as closely as possible. We have created a JIRA ticket to track this internally. 
Let me know what you think @glnro.